doc/user/migrating-existing-apis.md

```diff
@@ -0,0 +1,384 @@
+          required:
+          - size
+          type: object
+        status:
+          properties:
+            nodes:
+              items:
+                type: string
+              type: array
+          required:
+          - nodes
+          type: object
+  version: v1
+  subresources:
+    status: {}
+```
+
+Steps to upgrade the above CRD:
+
+1. Rename your CRD manifest file from `deploy/crds/operators_v1_catalogsourceconfig_crd.yaml` to `deploy/crds/catalogsourceconfigs.operators.coreos.com_catalogsourceconfigs_crd.yaml`
+
+    ```console
+    $ mv deploy/crds/cache_v1alpha1_memcached_crd.yaml deploy/crds/catalogsourceconfigs.operators.coreos.com_catalogsourceconfigs_crd.yaml
+    ```
+
+1. Create a `spec.versions` list that contains two elements for each version that now exists (`v1` and `v2`):
+
+    ```yaml
+    spec:
+      ...
+      # version is now v2, as it must match the first element in versions.
+      version: v2
+      versions:
+      - name: v2
+        # Set to true for this CRD version to be enabled in-cluster.
+        served: true
+        # Exactly one CRD version should be a storage version.
+        storage: true
+      - name: v1
+        served: true
+        storage: false
+    ```
+
+    The first version in `spec.versions` *must* match that in `spec.version` if `spec.version` exists in the manifest.
+
+1. *Optional:* `spec.versions` elements have a `schema` field that holds a version-specific OpenAPIV3 validation block to override the global `spec.validation` block. `spec.validation` will be used by the API server to validate one or more versions in `spec.versions` that do not have a `schema` block. If all versions have the same schema, leave `spec.validation` as-is and skip to the [following section](#golang-api-migrations-types-and-commonalities). If your CRD versions differ in scheme, copy `spec.validation` YAML to the `schema` field in each `spec.versions` element, then modify as needed:
+
+    ```yaml
+    spec:
+      ...
+      version: v2
+      versions:
+      - name: v2
+        served: true
+        storage: true
+        schema: # v2-specific OpenAPIV3 validation block.
+          openAPIV3Schema:
+            properties:
+              apiVersion:
+                type: string
+          ...
+      - name: v1
+        served: true
+        storage: false
+        schema: # v1-specific OpenAPIV3 validation block.
+          openAPIV3Schema:
+            properties:
+              apiVersion:
+                type: string
+          ...
+    ```
+
+    The API server will validate each version by its own `schema` if the global `spec.validation` block is removed. No validation will be performed if a `schema` does not exist for a version and `spec.validation` does not exist.
+
+    If the CRD targets a Kubernetes 1.13+ cluster with the `CustomResourceWebhookConversion` feature enabled, converting between multiple versions can be done using a [conversion webhooks][crd-conv-webhook].
```

Comment:
Should we mention the default "conversion", which just changes the GVK version field and nothing else?